heartbeat/beater/heartbeat.go

```diff
@@ -124,6 +124,13 @@ func (bt *Heartbeat) Run(b *beat.Beat) error {
 		defer bt.autodiscover.Stop()
 	}
 
+	// Configure the beats Manager to start after all the reloadable hooks are initialized
+	// and shutdown when the function return.
+	if err := b.Manager.Start(); err != nil {
+		return err
+	}
+	defer b.Manager.Stop()
```

Comment:
I think the defer here is going to place this in the wrong order. Because of the defer execution, the manager will be stopped before the `bt.monitorReloader.Stop()` and `defer bt.autodiscover.Stop()`. I think we should try to make sure that the manager is stopped always last.